src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -61,18 +63,23 @@ public ImageTagHelper(IHostingEnvironment hostingEnvironment, IMemoryCache cache
 
         protected IMemoryCache Cache { get; }
 
+        protected IUrlHelper UrlHelper { get; }
+
         /// <inheritdoc />
         public override void Process(TagHelperContext context, TagHelperOutput output)
         {
             if (AppendVersion)
             {
                 EnsureFileVersionProvider();
+                Src = UrlHelper.Content(Src.Trim());
                 output.Attributes[SrcAttributeName] = _fileVersionProvider.AddFileVersionToPath(Src);
             }
             else
             {
                 // Pass through attribute that is also a well-known HTML attribute.
                 output.CopyHtmlAttribute(SrcAttributeName, context);
+                output.Attributes[SrcAttributeName].Value =
+                    UrlHelper.Content(output.Attributes[SrcAttributeName].Value.ToString().Trim());
```

Comment:
If the value is an `HtmlString` (likely because an earlier tag helper set the value), this will cause double-encoding. Need a special case to avoid this.